From 885c1c605bd0050875a822729dfdbeae5a9a5e96 Mon Sep 17 00:00:00 2001
From: Scott Moreau <oreaus@gmail.com>
Date: Fri, 11 Jun 2021 09:27:46 -0600
Subject: [PATCH] gl/wayland: Use consistent wl_display when creating work
 queue for proxy wrapper

Without this, glimagesink since wayland 727c7903 fails with
gst-launch-1.0: ../src/wayland-client.c:2181: wl_proxy_set_queue:
Assertion 'proxy->display == queue->display' failed.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-plugins-base/-/merge_requests/1200>
---
 gst-libs/gst/gl/wayland/gstglwindow_wayland_egl.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/gst-libs/gst/gl/wayland/gstglwindow_wayland_egl.c b/gst-libs/gst/gl/wayland/gstglwindow_wayland_egl.c
index bb8be2756..ecfa03b49 100644
--- a/gst-libs/gst/gl/wayland/gstglwindow_wayland_egl.c
+++ b/gst-libs/gst/gl/wayland/gstglwindow_wayland_egl.c
@@ -561,8 +561,7 @@ gst_gl_window_wayland_egl_open (GstGLWindow * window, GError ** error)
    * each wayland resource we create as well as removing a race between
    * creation and the `wl_proxy_set_queue()` call. */
   window_egl->display.display = wl_proxy_create_wrapper (display->display);
-  window_egl->window.queue =
-      wl_display_create_queue (window_egl->display.display);
+  window_egl->window.queue = wl_display_create_queue (display->display);
 
   wl_proxy_set_queue ((struct wl_proxy *) window_egl->display.display,
       window_egl->window.queue);
-- 
GitLab

